tools/build_script_generator/cmake/module.lb

```diff
@@ -0,0 +1,149 @@
+    module.add_option(
+        StringOption(name="modm.namespace", default="robot",
+                     description="Namespace of the generated Modm communications"))
+
+    module.add_option(
+        BooleanOption(name="info.git", default=False,
+                      description="Generate CPP defines about the git repository state"))
+    module.add_option(
+        BooleanOption(name="info.build", default=False,
+                      description="Generate CPP defines about the build state"))
+
+    if options[":target"].identifier["platform"] == "avr":
+        # we need the clock:f_cpu option!
+        module.depends(":platform:clock")
+    return True
+
+
+def build(env):
+    pass
+
+
+def post_build(env, buildlog):
+    target = env["modm:target"]
+    platform = target.identifier["platform"]
+    family = target.identifier["family"]
+
+    build_tools = []
+    if len(env["modm.source"]): build_tools.append("system_design");
+    if len(env["image.source"]): build_tools.append("bitmap");
+    if env["info.git"] or env["info.build"]: build_tools.append("info");
+    if env["is_unittest"]: build_tools.append("unittest")
+
+    # Extract all source code files
+    files_to_build = []
+    outpath = env.outpath(".")
+    for operations in buildlog:
+        filename = os.path.relpath(operations.filename_out, outpath)
+        _, extension = os.path.splitext(filename)
+
+        if extension in [".c", ".cpp", ".cc", ".sx", ".S"]:
+            files_to_build.append(os.path.normpath(filename))
+    files_to_build.sort()
+
+    core_driver = target.get_driver("core")
+    core = core_driver["type"]
+    if core.startswith("cortex-m4"):
+        core = "cortex-m4"
+    elif core.startswith("cortex-m7"):
+        core = "cortex-m7"
+
+    memories = []
+    if "memory" in core_driver:
+        memories.extend([
+            {
+                k:(int(v) if v.isdigit() else (int(v, 16) if v.startswith("0x") else v))
+                for k, v in memory.items()
+            }
+            for memory in core_driver["memory"]
+        ])
+
+    env.substitutions = {
+        "partname": target.partname,
+        "target": target,
+        "family": family,
+        "platform": platform,
+        "project_path": os.getcwd(),
+        "compiler": "llvm" if family == "darwin" else "gcc",
+        "core": core,
+        "files": files_to_build,
+        "metadata": buildlog.metadata,
+        "openocd_config" : buildlog.metadata["openocd.configfile"],
+        "memories": memories,
+        "build_tools": build_tools,
+        "info_build": env["info.build"],
+        "info_git": env["info.git"],
+        "build_path": env["build.path"],
+        "project_name": env["project.name"],
+        "image_source": env["image.source"],
+        "modm_source": env["modm.source"],
+        "modm_container": env["modm.container"],
+        "modm_generate": env["modm.generate"],
+        "modm_namespace": env["modm.namespace"],
```

Comment:
Can you remove everything in this file that's not needed for the CMake generator?
It makes it easier for me to maintain this, if it's just the bare minimum functionality.

Reply:
Done